src/backend/bs_publish

```diff
@@ -2440,3 +2445,92 @@ $conf->{'getflavor'} = $BSConfig::publish_getflavor if $BSConfig::publish_getfla
 
 BSStdRunner::run('publisher', \@ARGV, $conf);
 
+
+=head2 check_for_container
+
+check if build result is valid container and upload if configured
+
+=cut
+
+sub check_for_container {
+  my ($r, $p, $file, $projid, $repoid, $arch) = @_;
+
+
+  # jump out if no config
+  return if (ref($BSConfig::docker_registry) ne "HASH");
+
+  # set default and check config for docker registry
+  my $dr = $BSConfig::docker_registry;
+  ## set default port
+  $dr->{port} ||= 443;
+
+  # check if minimal config items are set
+  if ( ! $dr->{host} or ! $dr->{user} or ! $dr->{password} ) {
+    BSUtil::printlog(
+      "No valid config found for docker registry: ".
+      "$dr->{host}/$dr->{user}/$dr->{password} (host/user/password)"
+    );
+    return;
+  }
+
+  # check if containerinfo is valid and get info for container
+  my $containerinfo = $p."containerinfo";
+  BSUtil::printlog("Checking containerinfo for $r/$containerinfo");
+  # FIXME: packid is needed as 3rd param
+  my $info = BSRepServer::Containerinfo::containerinfo2obsbinlnk($r, $containerinfo,'');
+
+  # TODO: should be more general to implement own upload
+  # prepare url for skopeo
+  my $name = $info->{name};
+  $name =~ s/^(?:container\:)?(.*)$/$1/;
+  $name = lc($name);
+
+  #TODO: remove debug log
+  BSUtil::printlog("projid/repoid/arch/name: $projid, $repoid, $arch, $name");
+
+  my $delimiter       = $dr->{repository_delimiter} || '/';
+  my $repository_base = $dr->{repository_base} || '/';
+  $projid =~ s/:/$delimiter/g;
+  my $tag = 'latest';
+  my $up = $dr->{uploader} || 'skopeo';
+  if ($up eq 'skopeo') {
+    my $dst = 'docker://'.$dr->{host}.':'.$dr->{port}.lc($repository_base.$projid."/$repoid/$arch/$name:$tag");
+
+    my $src = "$r/$file";
+    my $tempfile = decompress_container($src);
+    my @cmd = ("skopeo", "copy", "--dest-creds", "$dr->{user}:$dr->{password}", "docker-archive:$tempfile", $dst);
+    BSUtil::printlog("Uploading: '@cmd'");
+    die "Error while uploading\n" if qsystem(@cmd) ) {
+
+    unlink($tempfile);
```

Comment:
syntax problem, please do also the unlink always independend of success or failure.